### PR TITLE
refactor: ts-cli commands and flags with cobra

### LINKED ISSUE #297: refactoring of ts-cli's command line parameters using cobra
### Describe your question

to support nested subcommands help

```bash
ts-cli
ts-cli import ...
ts-cli export ...
ts-cli execute ...
ts-cli version
```

Only display the help information for the corresponding subcommand.

ref: https://cobra.dev/

### PR DESCRIPTION
**Pull request by xmh1011**
### What problem does this PR solve?

Issue Number: resolve #297 

### What is changed and how it works?
```
➜  openGemini git:(cobra) ./build/ts-cli -h        
ts-cli is a CLI tool of openGemini.

Usage:
  ts-cli [flags]
  ts-cli [command]

Available Commands:
  execute     Execute a query
  help        Help about any command
  import      Import data to openGemini
  interactive Work in interactive mode
  version     Display the version of openGemini CLI

Flags:
      --database string    Database to connect to openGemini.
  -h, --help               help for ts-cli
      --host string        ts-sql host to connect to. (default "localhost")
  -p, --password string    Password to connect to openGemini.
      --port int           ts-sql tcp port to connect to. (default 8086)
      --precision string   Specify the format of the timestamp: rfc3339, h, m, s, ms, u or ns. (default "ns")
      --socket string      openGemini unix domain socket to connect to.
      --ssl                Use https for connecting to openGemini.
      --unsafeSsl          Ignore ssl verification when connecting openGemini by https. (default true)
  -u, --username string    Username to connect to openGemini.

Use "ts-cli [command] --help" for more information about a command.
➜  openGemini git:(cobra) ./build/ts-cli import -h
Import line protocol text file to openGemini

Usage:
  ts-cli import [flags]

Examples:

$ ts-cli import --path=line_protocol_file.txt --host=127.0.0.1 --port=8086 --precision=s

Flags:
  -h, --help               help for import
      --host string        ts-sql host to connect to. (default "localhost")
  -p, --password string    Password to connect to openGemini.
      --path string        Path to the file to import.
      --port int           ts-sql tcp port to connect to. (default 8086)
      --precision string   Specify the format of the timestamp: rfc3339, h, m, s, ms, u or ns. (default "ns")
      --ssl                Use https for connecting to openGemini.
  -u, --username string    Username to connect to openGemini.
➜  openGemini git:(cobra) ./build/ts-cli execute -h
Execute a query

Usage:
  ts-cli execute [flags]

Examples:

$ ts-cli execute "show databases"
$ ts-cli execute "select xxx"


Flags:
  -h, --help   help for execute
➜  openGemini git:(cobra) ./build/ts-cli version -h
openGemini version info:
ts-cli: bff56572feec0722e88da2eaaf875efb5aaf322b
git: cobra bff56572feec0722e88da2eaaf875efb5aaf322b
os: darwin
arch: arm64
Display the version of openGemini CLI.

Usage:
  ts-cli version [flags]

Flags:
  -h, --help   help for version
```

### How Has This Been Tested?

- [x] Test cases to be added

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -137,6 +137,7 @@ require (
 	github.com/smartystreets/assertions v1.2.0 // indirect
 	github.com/spf13/cast v1.4.1 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
+	github.com/stretchr/objx v0.4.0 // indirect
```
Comment (shilinlee) [inline go.mod:140]:
```suggestion
```

File: app/ts-cli/cmd/root.go [hunk 2]
```diff
@@ -26,27 +28,60 @@ const (
...
+func Execute() {
+	rootCmd.AddCommand()
+	err := rootCmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+	rootCmd.Flags().StringVar(&options.Host, "host", DEFAULT_HOST, "openGemini host to connect to.")
```
Comment (shilinlee) [inline app/ts-cli/cmd/root.go:47]:
```suggestion
	rootCmd.Flags().StringVar(&options.Host, "host", DEFAULT_HOST, `ts-sql host to connect to. (default "localhost")`)
```

File: app/ts-cli/cmd/root.go [hunk 3]
```diff
@@ -26,27 +28,60 @@ const (
...
+	rootCmd.Flags().IntVar(&options.Port, "port", DEFAULT_PORT, "openGemini port to connect to.")
```
Comment (shilinlee) [inline app/ts-cli/cmd/root.go:48]:
```suggestion
	rootCmd.Flags().IntVar(&options.Port, "port", DEFAULT_PORT, "ts-sql tcp port to connect to. (default 8086)")
```

File: app/ts-cli/cmd/root.go [hunk 4]
```diff
@@ -26,27 +28,60 @@ const (
...
+	rootCmd.Flags().StringVar(&options.UnixSocket, "socket", "", "openGemini unix domain socket to connect to.")
+	rootCmd.Flags().StringVarP(&options.Username, "username", "u", "", "Username to connect to openGemini.")
+	rootCmd.Flags().StringVarP(&options.Password, "password", "p", "", "Password to connect to openGemini.")
+	rootCmd.Flags().StringVar(&options.Database, "database", "", "Database to connect to openGemini.")
+	rootCmd.Flags().BoolVar(&options.Ssl, "ssl", false, "Use https for connecting to openGemini.")
+	rootCmd.Flags().BoolVar(&options.IgnoreSsl, "unsafeSsl", true, "Ignore ssl verification when connecting openGemini by https.")
+	rootCmd.Flags().StringVar(&options.Precision, "precision", "ns", "Precision specifies the format of the timestamp:  rfc3339,h,m,s,ms,u or ns.")
```
Comment (shilinlee) [inline app/ts-cli/cmd/root.go:55]:
```suggestion
	rootCmd.Flags().StringVar(&options.Precision, "precision", "ns", "Specify the format of the timestamp: rfc3339, h, m, s, ms, u or ns. default is ns.")
```

File: app/ts-cli/cmd/root.go [hunk 5]
```diff
@@ -26,27 +28,60 @@ const (
...
+}
+
 var (
 	rootCmd = &cobra.Command{
 		Use:   "ts-cli",
 		Short: "openGemini CLI tool",
 		Long:  `ts-cli is a CLI tool of openGemini. It's compatible with influxdb cli.`,
```
Comment (shilinlee) [inline app/ts-cli/cmd/root.go:62]:
```suggestion
		Long:  `ts-cli is a CLI tool of openGemini.`,
```

File: app/ts-cli/cmd/root.go (hunk 4 above)
Comment (shilinlee) [reply app/ts-cli/cmd/root.go:55]: Check for sub command `import` also.

File: app/ts-cli/cmd/import.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	err := importCmd.MarkFlagRequired("path")
+	if err != nil {
+		return
+	}
+}
+
+var importCmd = &cobra.Command{
+	Use:   "import",
+	Short: "Import data to openGemini.",
+	Long:  `Import a previous database export from file.`,
```
Comment (shilinlee) [inline app/ts-cli/cmd/import.go:42]:
```suggestion
	Long:  `Import line protocol text file to openGemini.`,
```

File: app/ts-cli/cmd/version.go [hunk 7]
```diff
@@ -18,21 +18,30 @@ package cmd
...
 	"github.com/spf13/cobra"
 )
 
+var (
+	TsVersion = "v1.1.0rc0"
+	TsCommit  string
+	TsBranch  string
+)
+
+const TsStore = "TsCli"
```
Comment (shilinlee) [inline app/ts-cli/cmd/version.go:33]:
```suggestion
const TsCli = "TsCli"
```

File: app/ts-cli/cmd/version.go [hunk 8]
```diff
@@ -18,21 +18,30 @@ package cmd
...
 	rootCmd.AddCommand(versionCmd)
 }
 
 var versionCmd = &cobra.Command{
 	Use:   "version",
-	Short: "Print the version of openGemini CLI",
-	Long: `The version of openGemini CLI. Higher
-version of openGemini CLI is compatible with lower
- version of open-Gemini.`,
+	Short: "Print the version of openGemini CLI.",
```
Comment (shilinlee) [inline app/ts-cli/cmd/version.go:41]:
```suggestion
	Short: "Display the version of openGemini CLI.",
```

File: app/ts-cli/cmd/version.go [hunk 9]
```diff
@@ -18,21 +18,30 @@ package cmd
...
+	Long:  `The version of openGemini CLI. Higher version of openGemini CLI is compatible with lower version of open-Gemini.`,
```
Comment (shilinlee) [inline app/ts-cli/cmd/version.go:42]:
```suggestion
	Long:  `Display the version of openGemini CLI.`,
```

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: app/ts-cli/cmd/import.go [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+	importCmd.Flags().StringVar(&options.Path, "path", "", "Path to the file to import.")
+	err := importCmd.MarkFlagRequired("path")
+	if err != nil {
+		return
+	}
+}
+
+var importCmd = &cobra.Command{
+	Use:   "import",
+	Short: "Import data to openGemini.",
```
Comment (shilinlee) [inline app/ts-cli/cmd/import.go:41]:
```suggestion
	Short: "Import data to openGemini",
```

**Review by shilinlee — CHANGES_REQUESTED** [review]

**Review by Shoothzj — APPROVED** [review]

File: app/ts-cli/main.go [hunk 11]
```diff
@@ -17,34 +17,9 @@ limitations under the License.
...
-
-	parse "github.com/influxdata/influxdb/cmd"
-	"github.com/openGemini/openGemini/app"
 	"github.com/openGemini/openGemini/app/ts-cli/cmd"
 )
 
-var (
-	TsVersion = "v1.1.0rc0"
-	TsCommit  string
-	TsBranch  string
```
Comment (shilinlee) [inline app/ts-cli/main.go:32]: Version info keep at `main` package.

**Review by shilinlee — CHANGES_REQUESTED** [review]

**Review by shilinlee — APPROVED** [review]

**Review by scuzyj — APPROVED** [review]

**Pull request merged**